ocp_utilities/operators.py

```diff
@@ -334,14 +346,39 @@ def _icsp(_repository_digest_mirrors):
         namespace="openshift-config",
     )
 
-    catalog_source = CatalogSource(
+    iib_catalog_source = create_catalog_source_from_image(
         name=name,
         namespace=operator_market_namespace,
-        display_name=name,
         image=_iib_index_image,
+    )
+    return iib_catalog_source
+
+
+def create_catalog_source_from_image(
+    name, namespace, image, source_type=None, update_strategy_registry_poll_interval=None
+):
+    """
+    Create CatalogSource for given image
+
+    Args:
+        name (str): Name for the catalog source (used in 'name, display_name and publisher').
+        image (str): Image index for the catalog.
+        namespace (str): Namespace where CatalogSource will be created.
+        source_type (str, optional): Name of the source type.
+        update_strategy_registry_poll_interval (str, optional): Time interval between checks of the latest
+                catalog_source version.
+
+    Returns:
+        CatalogSource: catalog source object.
+    """
+    catalog_source = CatalogSource(
```

Comment:
please add `client=admin_client` 
whenever working with ocp resources we want to make sure we have a client with priviledges

Reply:
Added